lab-katherine/src/main.js

```diff
@@ -0,0 +1,161 @@
+          // redditStuff[res.body.data.children[n].data.author] = res.body.data.children[n].data.title;
+          return redditStuff;
+          // lookup[n] = res.body.data.children[n].data.title;
+          // let author = res.body.data.children[n].data.author;
+        });
+        console.log(boardLookup);
+        try {
+          localStorage.boardLookup = JSON.stringify(boardLookup);
+          console.log(localStorage.boardLookup)
+          this.setState({boardLookup});
+        } catch(e) {
+          console.error(e);
+        }
+      })
+  }
+
+  handleChange(e) {
+    this.setState({ searchFormBoard: e.target.value});
+  }
+  render() {
+    return (
+      <form
+        className="reddit-form"
+        onSubmit={this.handleSubmit}>
+
+        <input
+          type="text"
+          name="boardName"
+          placeholder="search"
+          value={this.state.searchFormBoard}
+          onChange={this.handleChange} />
+      </form>
+    );
+  }
+}
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      boardLookup: {},
+      boardSelected: null,
+      boardNameError: null,
+    };
+    this.boardAppSelect = this.boardAppSelect.bind(this);
+  }
+  componentDidUpdate() {
+    console.log('__STATE__', this.state);
+  }
+  // componentDidMount() {
+  //   if(localStorage.boardLookup) {
+  //     try {
+  //       let boardLookup = JSON.parse(localStorage.boardLookup);
+  //       this.setState({boardLookup});
+  //     } catch(e) {
+  //       console.error(e);
+  //     }
+  //   } else {
+  //     superagent.get(`${API_URL}.json?limit=20`)
+  //       .then(res => {
+  //         console.log('body', res.body.data);
+  //         console.log(res.body.data.children)
+  //         let boardLookup = res.body.data.children.reduce((lookup, n) =>{
+  //           console.log(n[0])
+  //           lookup[n.name] = n.subreddit;
+  //           return lookup;
+  //         }, {});
+  //         try{
+  //           localStorage.boardLookup = JSON.stringify(boardLookup);
+  //           this.setState({boardLookup});
+  //         } catch(e) {
+  //           console.error(e);
+  //         }
+  //       })
+  //       .catch(console.error);
+  //   }
+  // }
+  boardAppSelect(){
+    console.log('boardAppSelect', this.state.boardLookup)
+    if(!this.state.boardLookup[title]) {
+      this.setState({
+        boardSelected: null,
+        boardNameError: name,
+      });
+    } else {
+      console.log(this.state.boardLookup[title]);
+      superagent.get(this.state.boardLookup[title])
+        .then(res => {
+          this.setState({
+            boardSelected: res.body,
+            boardNameError: null,
+          });
+        })
+        .catch(console.error);
+    }
+  }
+
+  render() {
+    return (
+      <section className="application">
+        <h1>Reddit Search</h1>
+        <RedditForm
+          boardSelect={this.boardAppSelect}
+          scott='hello world'/>
```

Comment:
you should be building this app out from scratch, not copying and pasting from the demo. That won't help you learn.